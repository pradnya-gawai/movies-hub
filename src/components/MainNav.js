import React, { useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import BottomNavigation from '@material-ui/core/BottomNavigation';
import BottomNavigationAction from '@material-ui/core/BottomNavigationAction';
import TvIcon from '@material-ui/icons/Tv';
import MovieIcon from '@material-ui/icons/Movie';
import SearchIcon from '@material-ui/icons/Search';
import WhatshotIcon from '@material-ui/icons/Whatshot';

const useStyles = makeStyles({
  root: {
    width: '100%',
    position: 'fixed',
    bottom: 0,
    backgroundColor: '#2d313a',
    zIndex: 100,
  },
});
export default function MainNav() {
  const classes = useStyles();
  const [value, setValue] = React.useState(0);
  const history = useHistory();

  useEffect(() => {
    if (value === 0) {
      history.push('/');
    } else if (value === 1) {
      history.push('/movies');
    } else if (value === 2) {
      history.push('/series');
    } else if (value === 3) {
      history.push('/search');
    }
  }, [value, history]);

  return (
    <div data-testid="main_nav">
      <BottomNavigation
        data-testid="mui-btm-nav"
        value={value}
        onChange={(event, newValue) => {
          setValue(newValue);
        }}
        showLabels
        className={classes.root}
      >
        <BottomNavigationAction
          data-testid="whatshot"
          style={{ color: 'white' }}
          label="Trending"
          icon={<WhatshotIcon />}
        />
        <BottomNavigationAction
          data-testid="movie-icon"
          style={{ color: 'white' }}
          label="Movies"
          icon={<MovieIcon />}
        />
        <BottomNavigationAction
          data-testid="tv-icon"
          style={{ color: 'white' }}
          label="TV Series"
          icon={<TvIcon />}
        />
        <BottomNavigationAction
          data-testid="search-icon"
          style={{ color: 'white' }}
          label="Search"
          icon={<SearchIcon />}
        />
      </BottomNavigation>
    </div>
  );
}
