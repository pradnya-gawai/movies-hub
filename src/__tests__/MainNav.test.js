import MainNav from "../components/MainNav";
import { render, screen } from "@testing-library/react";
import { unmountComponentAtNode } from "react-dom";
import { BrowserRouter } from "react-router-dom";
// import { act } from "react-test-renderer";
const MockMainNav = () => {
  return (
    <BrowserRouter>
      <MainNav />
    </BrowserRouter>
  );
};

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
  // render(<MockMainNav/>)
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

// describe("MainNav", () => {
//   test("should render  main nagivation correctly", () => {
//     const wrapper = shallow(<MainNav />);
//     expect(wrapper).toMatchSnapshot();
//   });
// });
it("renders correctly again", () => {
    const wrapper = render(
      <MockMainNav />
    );
  
    expect(wrapper).toMatchSnapshot();
  });
  
describe("check icon", () => {
  test("check icon correctly", () => {
    const { getByTestId } = render(<MockMainNav />);
    expect(getByTestId("whatshot").textContent).toMatch("Trending");
    expect(getByTestId("movie-icon").textContent).toMatch("Movies");
    expect(getByTestId("tv-icon").textContent).toMatch("TV Series");
    expect(getByTestId("search-icon").textContent).toMatch("Search");
  });
});
// test("should disply whatshot icon", () => {
//   render(<MockMainNav />);
//   const whatshotElement = screen.getByTestId("whatshot");
//   expect(whatshotElement).toBeInTheDocument();
// });

// test("should disply movie icon", () => {
//   render(<MockMainNav />);
//   const whatshotElement = screen.getByTestId("movie-icon");
//   expect(whatshotElement).toBeInTheDocument();
// });

// test("should disply tv-icon", () => {
//   render(<MockMainNav />);
//   const whatshotElement = screen.getByTestId("tv");
//   expect(whatshotElement).toBeInTheDocument();
// });
// test("should display search icon", () => {
//   render(<MockMainNav />);
//   const whatshotElement = screen.getByTestId("search-icon");
//   expect(whatshotElement).toBeInTheDocument();
// // });
// test("test state", () => {
//     const wrapper =shallow(<MainNav />);
//     wrapper.find('#movie-icon').simulate('change')
//     expect(wrapper.find)
// })   