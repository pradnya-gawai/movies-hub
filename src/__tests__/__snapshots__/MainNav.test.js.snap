// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly again 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div>
      <div
        data-testid="main_nav"
      >
        <div
          class="MuiBottomNavigation-root makeStyles-root-1"
          data-testid="mui-btm-nav"
        >
          <button
            class="MuiButtonBase-root MuiBottomNavigationAction-root Mui-selected"
            data-testid="whatshot"
            style="color: white;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiBottomNavigationAction-wrapper"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M13.5.67s.74 2.65.74 4.8c0 2.06-1.35 3.73-3.41 3.73-2.07 0-3.63-1.67-3.63-3.73l.03-.36C5.21 7.51 4 10.62 4 14c0 4.42 3.58 8 8 8s8-3.58 8-8C20 8.61 17.41 3.8 13.5.67zM11.71 19c-1.78 0-3.22-1.4-3.22-3.14 0-1.62 1.05-2.76 2.81-3.12 1.77-.36 3.6-1.21 4.62-2.58.39 1.29.59 2.65.59 4.04 0 2.65-2.15 4.8-4.8 4.8z"
                />
              </svg>
              <span
                class="MuiBottomNavigationAction-label Mui-selected"
              >
                Trending
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiBottomNavigationAction-root"
            data-testid="movie-icon"
            style="color: white;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiBottomNavigationAction-wrapper"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 4l2 4h-3l-2-4h-2l2 4h-3l-2-4H8l2 4H7L5 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V4h-4z"
                />
              </svg>
              <span
                class="MuiBottomNavigationAction-label"
              >
                Movies
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiBottomNavigationAction-root"
            data-testid="tv-icon"
            style="color: white;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiBottomNavigationAction-wrapper"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21 3H3c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h5v2h8v-2h5c1.1 0 1.99-.9 1.99-2L23 5c0-1.1-.9-2-2-2zm0 14H3V5h18v12z"
                />
              </svg>
              <span
                class="MuiBottomNavigationAction-label"
              >
                TV Series
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiBottomNavigationAction-root"
            data-testid="search-icon"
            style="color: white;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiBottomNavigationAction-wrapper"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
              <span
                class="MuiBottomNavigationAction-label"
              >
                Search
              </span>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="main_nav"
    >
      <div
        class="MuiBottomNavigation-root makeStyles-root-1"
        data-testid="mui-btm-nav"
      >
        <button
          class="MuiButtonBase-root MuiBottomNavigationAction-root Mui-selected"
          data-testid="whatshot"
          style="color: white;"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiBottomNavigationAction-wrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M13.5.67s.74 2.65.74 4.8c0 2.06-1.35 3.73-3.41 3.73-2.07 0-3.63-1.67-3.63-3.73l.03-.36C5.21 7.51 4 10.62 4 14c0 4.42 3.58 8 8 8s8-3.58 8-8C20 8.61 17.41 3.8 13.5.67zM11.71 19c-1.78 0-3.22-1.4-3.22-3.14 0-1.62 1.05-2.76 2.81-3.12 1.77-.36 3.6-1.21 4.62-2.58.39 1.29.59 2.65.59 4.04 0 2.65-2.15 4.8-4.8 4.8z"
              />
            </svg>
            <span
              class="MuiBottomNavigationAction-label Mui-selected"
            >
              Trending
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiBottomNavigationAction-root"
          data-testid="movie-icon"
          style="color: white;"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiBottomNavigationAction-wrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 4l2 4h-3l-2-4h-2l2 4h-3l-2-4H8l2 4H7L5 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V4h-4z"
              />
            </svg>
            <span
              class="MuiBottomNavigationAction-label"
            >
              Movies
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiBottomNavigationAction-root"
          data-testid="tv-icon"
          style="color: white;"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiBottomNavigationAction-wrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M21 3H3c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h5v2h8v-2h5c1.1 0 1.99-.9 1.99-2L23 5c0-1.1-.9-2-2-2zm0 14H3V5h18v12z"
              />
            </svg>
            <span
              class="MuiBottomNavigationAction-label"
            >
              TV Series
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiBottomNavigationAction-root"
          data-testid="search-icon"
          style="color: white;"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiBottomNavigationAction-wrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
            <span
              class="MuiBottomNavigationAction-label"
            >
              Search
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
